test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
Got it. I created an expect_val_eq function to do this and just pushed the changes.